lasy/profiles/from_openpmd_profile.py

```diff
@@ -81,128 +74,32 @@ def __init__(
-                it = s.iterations[iteration]
-                omg0 = it.meshes["laserEnvelope"].get_attribute("angularFrequency")
-                wavelength = 2 * np.pi * c / omg0
-                array = F
-
             axes_order = ["x", "y", "t"]
 
         else:  # Envelope assumes axial symmetry processing RZ data
-            assert m.axes in [
-                {0: "r", 1: "z"},
-                {0: "z", 1: "r"},
-                {0: "r", 1: "t"},
-                {0: "t", 1: "r"},
-            ]
-
             dim = "rt"
-
-            if m.axes in [{0: "z", 1: "r"}, {0: "t", 1: "r"}]:
-                F = F.swapaxes(0, 1)
-
-            if "z" in m.axes.values():
-                t = (m.z - m.z[0]) / c
-            else:
-                t = m.t
-            r = m.r[m.r.size // 2 :]
-            axes = {"r": r, "t": t}
-
-            F = 0.5 * (
-                F[F.shape[0] // 2 :, :] + np.flip(F[: F.shape[0] // 2, :], axis=0)
-            )
-
             if phase_unwrap_1d is None:
                 phase_unwrap_1d = False
-            # If array does not contain the envelope but the electric field,
-            # extract the envelope with a Hilbert transform
-            if not envelope:
-                # Assumes z is last dimension!
-                h = hilbert(F)
-
-                if "z" in m.axes.values():
-                    # Flip to get complex envelope in t assuming z = -c*t
-                    h = np.flip(h, axis=-1)
-
-                # Get central wavelength from array
-                lo = (axes["r"][0], axes["t"][0])
-                hi = (axes["r"][-1], axes["t"][-1])
-                npoints = (axes["r"].size, axes["t"].size)
-                grid = Grid(dim, lo, hi, npoints, n_azimuthal_modes=1)
-                assert np.all(grid.axes[0] == axes["r"])
-                assert np.allclose(grid.axes[1], axes["t"], rtol=1.0e-14)
-                assert grid.field.shape == h[np.newaxis].shape
-                grid.field = h[np.newaxis]
-                omg_h, omg0_h = get_frequency(
-                    grid,
-                    dim=dim,
-                    is_envelope=False,
-                    is_hilbert=True,
-                    phase_unwrap_1d=phase_unwrap_1d,
-                )
-                wavelength = 2 * np.pi * c / omg0_h
-                array = h * np.exp(1j * omg0_h * t)
-            else:
-                s = io.Series(path + "/" + prefix + "_%T.h5", io.Access.read_only)
-                it = s.iterations[iteration]
-                omg0 = it.meshes["laserEnvelope"].get_attribute("angularFrequency")
-                wavelength = 2 * np.pi * c / omg0
-                array = F
-
             axes_order = ["r", "t"]
 
+        F, axes = reorder_array(F, m, dim)
+
+        # If array does not contain the envelope but the electric field,
+        # extract the envelope with a Hilbert transform
+        if not np.iscomplexobj(F):
```

Comment:
This assumes that `is_complex iff is_envelope`. This could be dangerous, and maybe a safer option would be to check if `envelopeField` exists and has the right value, as discussed [in the openPMD-standard PR](https://github.com/openPMD/openPMD-standard/blob/aeadcda9d71ff702d7cb412b04595f76b0d42105/EXT_LaserEnvelope.md)? Until this is in the standard and the codes have adopted it,  we could add the optional argument, which would overwrite the test `envelopeField` test. What do you think?